src/ray/core_worker/store_provider/store_provider.cc

```diff
@@ -0,0 +1,22 @@
+#include "arrow/buffer.h"
+#include "arrow/io/memory.h"
+
+#include "ray/core_worker/store_provider/store_provider.h"
+#include "ray/common/status.h"
+
+namespace ray {
+
+Status BufferedRayObject::WriteDataTo(std::shared_ptr<Buffer> buffer) const {
+  memcpy(buffer->Data(), data_->Data(), data_->Size());
```

Comment:
How about change the param to be  `std::shared_ptr<Buffer> *buffer` so that we can resize it inside this function? The current code assumes the passed in `buffer` has the same size with this object, which implies user must call `DataSize()` before calling this function. It seems to be easier to just let user pass in an address of a buffer, and let this function takes care of all the stuff.

Reply:
That won't work with the plasma store - it returns a fixed size buffer that metadata and data should be written into.

Reply:
ok, in that case could you please add a check before the `memcpy` to make sure the buffer passed in does have enough size, and return failure if it doesn't?